PWGJE/Tasks/jetHadronRecoil_OO.cxx

```diff
@@ -0,0 +1,541 @@
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processMC_DetLevel_Weighted, "process MC detector level with event weight", false);
+
+  void processMC_PartLevel(filtered_Coll_PartLevel const& collision,
+                           filtered_Jets_PartLevel const& jets,
+                           aod::JetParticles const& particles)
+  {
+    spectra.fill(HIST("vertexZ"), collision.posZ());
+    fillMCPHistograms(jets, particles);
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processMC_PartLevel, "process MC particle level", false);
+
+  void processMC_PartLevel_Weighted(filtered_Coll_PartLevel const& collision,
+                                    filtered_Jets_PartLevel const& jets,
+                                    aod::JetParticles const& particles)
+  {
+    auto weight = collision.weight();
+    spectra.fill(HIST("vertexZ"), collision.posZ(), weight);
+    fillMCPHistograms(jets, particles, weight);
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processMC_PartLevel_Weighted, "process MC particle level with event weight", false);
+
+  void processJetsMatched(filtered_Coll_DetLevel_to_GetWeight const& collision,
+                          aod::JetMcCollisions const&,
+                          filtered_Tracks const& tracks,
+                          filtered_MatchedJets_DetLevel const& mcdjets,
+                          filtered_MatchedJets_PartLevel const& mcpjets)
+  {
+    if (skipEvent(collision))
+      return;
+    auto mcpjetsPerMCCollision = mcpjets.sliceBy(PartJetsPerCollision, collision.mcCollisionId());
+    fillMatchedHistograms(tracks, mcdjets, mcpjetsPerMCCollision);
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processJetsMatched, "process matching of MC jets (no weight)", false);
+
+  void processJetsMatched_Weighted(filtered_Coll_DetLevel_to_GetWeight const& collision,
+                                   aod::JetMcCollisions const&,
+                                   filtered_Tracks const& tracks,
+                                   filtered_MatchedJets_DetLevel const& mcdjets,
+                                   filtered_MatchedJets_PartLevel const& mcpjets)
+  {
+    if (skipEvent(collision))
+      return;
+
+    auto mcpjetsPerMCCollision = mcpjets.sliceBy(PartJetsPerCollision, collision.mcCollisionId());
+    auto weight = collision.mcCollision().weight();
+
+    fillMatchedHistograms(tracks, mcdjets, mcpjetsPerMCCollision, weight);
+  }
+  PROCESS_SWITCH(jetHadronRecoil_OO, processJetsMatched_Weighted, "process matching of MC jets (weighted)", false);
+
+  //------------------------------------------------------------------------------
+  // Auxiliary functions
+  template <typename Collision>
+  bool skipEvent(const Collision& coll)
+  {
+    /// \TODO: Do need !jetderiveddatautilities::selectTrigger(collision, triggerMaskBits) flag to skip an event?
+    return !coll.selection_bit(aod::evsel::kNoTimeFrameBorder) || !jetderiveddatautilities::selectCollision(coll, eventSelection);
```

Comment:
hi Artem, isnt the NoTimeFrameBorder cut already included in sel8?
For this one selection this means that you are not fully using the derived data and are instead relying on the evsels table which we should find a way of avoiding. If the timeframe border cut is not included we can add this as a selection in the jet framework that can be accessed with jetderiveddatautilities::selectCollision

Reply:
If you could fix it now it would be great, because i think this way if you remove the need for it then you can remove the dependency on the Event selection tables and then you can run directly on the JE derived data